crates/sqlexec/src/planner/physical_plan/remote_scan.rs

```diff
@@ -31,6 +32,25 @@ pub enum ProviderReference {
     Provider(Arc<dyn TableProvider>),
 }
 
+impl PartialEq for ProviderReference {
+    fn eq(&self, other: &Self) -> bool {
+        match (self, other) {
+            (Self::RemoteReference(id), Self::RemoteReference(other_id)) => id.eq(other_id),
+            _ => false,
+        }
+    }
+}
+
+impl Eq for ProviderReference {}
+
+impl Hash for ProviderReference {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        if let Self::RemoteReference(id) = self {
+            id.hash(state)
+        }
+    }
+}
```

Comment:
Actually this seems kind of concerning. Do we need the hash.

Reply:
So it's due to the `LogicalPlan` serialization stuff. I had a branch where I removed all that but it got deleted. I am planning on removing all this. It's never used but exists because of trait constraints.